dataflowengine/src/main/scala/io/shiftleft/dataflowengine/layers/dataflows/DataFlowRunner.scala

```diff
@@ -8,9 +8,16 @@ import io.shiftleft.dataflowengine.semanticsloader.Semantics
 
 class DataFlowRunner(semantics: Semantics) {
 
-  def run(cpg: Cpg, serializedCpg: SerializedCpg): Unit = {
+  def run(cpg: Cpg, serializedCpg: SerializedCpg): Unit = run(cpg, Some(serializedCpg))
+
+  def run(cpg: Cpg, serializedCpg: Option[SerializedCpg] = None): Unit = {
     val enhancementExecList = List(new PropagateEdgePass(cpg, semantics), new ReachingDefPass(cpg))
-    enhancementExecList.foreach(_.createApplySerializeAndStore(serializedCpg))
+
+    if (serializedCpg.isDefined) {
```

Comment:
more idiomatic, but doesn't really matter / make a difference I guess..
```suggestion
    serializedCpg match {
      case Some(serializedCpg) => enhancementExecList.foreach(_.createApplySerializeAndStore(serializedCpg))
      case None =>                enhancementExecList.foreach(_.createAndApply)
    }
```